### ISSUE TITLE
Problem with small images

### ISSUE THREAD

**Issue by benjaminfunk** [marrone/grunt-tinypng#18]
If i just have 1 small image to compress the progress works well but the downloaded image was not downloaded complete. Seems that the seems didnt wait for complete?

The second attachments ist the damged file. If i upload the image directly to tinypng.com it works well.

![profile-background](https://cloud.githubusercontent.com/assets/856290/9465709/50a6a5de-4b2f-11e5-8d81-f82a7bdbf74f.png)

![profile-background](https://cloud.githubusercontent.com/assets/856290/9465766/b52291e4-4b2f-11e5-9029-ebb639efe67e.png)

**Comment by marrone** [comment]
![issue18 min](https://cloud.githubusercontent.com/assets/1319/9467763/e37bf752-4b0b-11e5-8f63-6a5247cf62f2.png)

hm, the attached image is what I get when I try it. Does your broken sample happen every time? What OS/environment are you running? Do you have the latest version of the plugin?

Could you provide the gruntfile config you are using the produces the broken image?

Thanks

**Comment by benjaminfunk** [comment]
Updated to v0.6.0 an problem still exist. And its happen every time. Sometime more damaged sometimes less. Windows 10 and Windows 8.1 with npm v2.10.1. grunt v.0.4.5.

After update: "graceful-fs" module not found. Only in "devDepencies" but still use in production?

Gruntfile.js

```javascript
tinypng:
		{
			options:
			{
				apiKey: '<%= application.config.api.tinypng %>',
				checkSigs: true,
				sigFile: '<%= application.destination.temp%>/tinypng_signatures.json',
				summarize: true,
				showProgress: true,
				stopOnImageError: true
			},
			
			compress:
			{
				expand: true, 
				cwd: '<%= application.source.image %>',
				src: ['**/*.{png,jpg,jpeg}'],
				dest: '<%= application.destination.image %>'
			}
		}
```

**Comment by benjaminfunk** [comment]
If i remove the Line 110 - 118 in the file /tasks/model/ImageProcess.js it works fine for me. Maybe this helps a bit :)

**Comment by netpro2k** [comment]
I am seeing an issue like this intermittently as well. Compressing the same file succeeds on one run, but might fail on another. As @benjaminfunk said, it does seem like the end callback is somehow being fired before the file download is complete.

**Comment by marrone** [comment]
Hey guys, sorry for this issue. Unfortunately, I cannot reproduce it on my machine. But just looking at what you guys describe is going on, I think I see a potential issue and applied a fix in this branch:
https://github.com/marrone/grunt-tinypng/tree/issue-18

Can someone try it out and tell me if that resolves the issue for you? 
Thanks

**Comment by benjaminfunk** [comment]
IT WORKS! :) Tested it on 2 different W10 machines with different images. No Problems!

**Comment by netpro2k** [comment]
Unfortunately I am unable to readily produce this issue to confirm if the fix is working or not.. That said, talking to my team it does sound like the machine we were seeing it on the most did not have an SSD which would be consistent with the proposed fix.

**Comment by benjaminfunk** [comment]
This could be the point. All my machines runs on a SSD

**Comment by netpro2k** [comment]
Are you planning to merge this fix into master soon? If not I'll just fork for use in our existing gruntfiles

**Comment by marrone** [comment]
It has been merged and 0.6.1 has been published to npm
